docs/Training-Curriculum-Learning.md

```diff
@@ -2,68 +2,97 @@
+curriculums are held in what we call a metacurriculum. A metacurriculum allows
+different brains to follow different curriculums within the same environment.
+
+### Specifying a Metacurriculum
+
+We first create a folder inside `python/curricula/` for the environment we want
+to use curriculum learning with. For example, if we were creating a metacurriculum
+for Wall Jump, we would create the folder `python/curricula/wall-jump/`. We will place
+our curriculums inside this folder.
+
+### Specifying a Curriculum
+
+In order to define a curriculum, the first step is to decide which
+parameters of the environment will vary. In the case of the Wall Jump environment, what
+varies is the height of the wall. We define this as a `Reset Parameter` in the Academy
+object of our scene, and by doing so it becomes adjustable via the Python API. Rather
+than adjusting it by hand, we will create a simple JSON file which describes the
+structure of the curriculum. Within it, we can specify which points in the training process
+our wall height will change, either based on the percentage of training steps which have
+taken place, or what the average reward the agent has received in the recent past is.
+Below is an example curriculum for the BigWallBrain in the Wall Jump environment.
 
 ```json
 {
     "measure" : "progress",
     "thresholds" : [0.1, 0.3, 0.5],
     "min_lesson_length" : 2,
-    "signal_smoothing" : true, 
-    "parameters" : 
+    "signal_smoothing" : true,
+    "parameters" :
     {
         "big_wall_min_height" : [0.0, 4.0, 6.0, 8.0],
-        "big_wall_max_height" : [4.0, 7.0, 8.0, 8.0],
-        "small_wall_height" : [1.5, 2.0, 2.5, 4.0]
+        "big_wall_max_height" : [4.0, 7.0, 8.0, 8.0]
     }
 }
 ```
 
 * `measure` - What to measure learning progress, and advancement in lessons by.
-    * `reward` - Uses a measure received reward. 
+    * `reward` - Uses a measure received reward.
     * `progress` - Uses ratio of steps/max_steps.
 * `thresholds` (float array) - Points in value of `measure` where lesson should be increased.
-* `min_lesson_length` (int) - How many times the progress measure should be reported before 
+* `min_lesson_length` (int) - How many times the progress measure should be reported before
 incrementing the lesson.
 * `signal_smoothing` (true/false) - Whether to weight the current progress measure by previous values.
     * If `true`, weighting will be 0.75 (new) 0.25 (old).
 * `parameters` (dictionary of key:string, value:float array) - Corresponds to academy reset parameters to control. Length of each array
 should be one greater than number of thresholds.
+
+
+Once our curriculum is defined, we have to use the reset parameters we defined and modify
+the environment from the agent's `AgentReset()` function. See
+[WallJumpAgent.cs](https://github.com/Unity-Technologies/ml-agents/blob/master/unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs)
+for an example.
```

Comment:
We should specify that if the academy does not reset, the reset parameters will not be updated